saleor/graphql/order/mutations/order_update_shipping.py

```diff
@@ -133,54 +114,19 @@ def perform_mutation(  # type: ignore[override]
                 "postal_code_rules"
             ),
         )
-        shipping_channel_listing = (
-            shipping_models.ShippingMethodChannelListing.objects.filter(
-                shipping_method=method, channel=order.channel
-            ).first()
-        )
-        if not shipping_channel_listing:
-            raise ValidationError(
-                {
-                    "shipping_method": ValidationError(
-                        "Shipping method not available in the given channel.",
-                        code=OrderErrorCode.SHIPPING_METHOD_NOT_APPLICABLE.value,
-                    )
-                }
-            )
+        shipping_channel_listing = cls.validate_shipping_channel_listing(method, order)
 
         shipping_method_data = convert_to_shipping_method_data(
             method,
             shipping_channel_listing,
         )
         manager = get_plugin_manager_promise(info.context).get()
-        clean_order_update_shipping(order, shipping_method_data, manager)
-
-        order.shipping_method = method
-        order.shipping_method_name = method.name
+        if order.status != OrderStatus.DRAFT:
```

Comment:
Can you explain, why we clean up the order shipping only for the orders that are not draft? It is not clear. Would be also good to probably left a comment here

Reply:
`clean_order_update_shipping` validates shipping method. We don't want to do that for drafts - this enables users to be more flexible with drafts (e.g. add shipping method to draft without address). This mutation can be used not only on draft orders (also unconfirmed) so we want to skip this validation for orders with status `DRAFT`